chore: update supported python versions in README to >= 3.6

rajatbhatta opened the pull request:
Update supported python versions in README to >= 3.6, in line with setup.py which indicates python_requires=">=3.6".

rajatbhatta committed:
chore: update supported python versions in README to >= 3.6

Update supported python versions in README to >= 3.6, in line with setup.py which indicates python_requires=">=3.6".

ansh0l commented on README.rst:
```diff
@@ -56,7 +56,7 @@ dependencies.
 Supported Python Versions
 ^^^^^^^^^^^^^^^^^^^^^^^^^
-Python >= 3.5
+Python >= 3.6
```
Would this mean all customers using python 3.5 face incompatibility with library?

rajatbhatta replied:
I think so. 

Support for Python 3.5 and earlier versions was removed as part of this commit: https://github.com/googleapis/python-spanner/pull/147/commits/606258bc728e5eed2b0b032a6428e1b77c39b3ce.

Adding @vi3k6i5 to comment/confirm the same.

IlyaFaer replied:
Users will get problems if they'll try to upgrade to the new version of the `python-spanner` package. See the setup param for details:
https://packaging.python.org/en/latest/guides/distributing-packages-using-setuptools/#python-requires

So, until they'll try to upgrade, they're okay.
Plus to this, I don't really think it's a big move to upgrade from 3.5 to 3.6. It's not jumping from 2 to 3, it's a comparatively small step, which, most likely, will not require manual work.

vi3k6i5 replied:
Makes sense. This is anyways just a documentation fix, the setup.py file was updated a while back.

vi3k6i5 replied:
Maybe we should add Python 3.5 to Deprecated Python Versions below ? @ansh0l @IlyaFaer ?

IlyaFaer replied:
Yeah, I guess, let's add

rajatbhatta replied:
This is done.

ansh0l replied:
sounds good

vi3k6i5 reviewed (approved):
LGTM

rajatbhatta committed:
chore: update the deprecated python versions in readme file

Python 3.5 has been deprecated. So, adding it to the list of deprecated versions.

ansh0l reviewed (approved):
lgtm

parthea committed:
Merge branch 'main' into update-readme-python-version
